Signals fire world.time

Gurkenglas opened the pull request:
## Why It's Good For The Game

All manner of shenanigans, and it's data that should be available. And isn't it neat how much you can do with a one line edit?

## Changelog
:cl:
expansion: Signals now fire a timestamp.
/:cl:

Gurkenglas committed:
Signals fire world.time

Mothblocks reviewed (changes requested):
`world.time` is in deciseconds, which is an engine detail that shouldn't be replicated outwards. Divide it by (1 SECONDS) before sending it so that players can use it more naturally.

Also, CC @Watermelon914 if they have any thoughts on this.

Watermelon914 commented:
sure this is fine, it gives more data and might make some more interesting things more possible

Gurkenglas committed:
wouldnt that have been centiseconds

Watermelon914 commented on code/__DEFINES/wiremod.dm:
```diff
@@ -49,7 +49,7 @@
 
 /// The value that is sent whenever a component is simply sending a signal. This can be anything.
-#define COMPONENT_SIGNAL 1
+#define COMPONENT_SIGNAL (world.time SECONDS)
```
don't do world.time SECONDS, just do world.time * 10.

This looks incredibly confusing to someone who might not understand it, and it looks really ugly in general.

Gurkenglas replied:
would you want all uses of SECONDS everywhere removed, then?

Watermelon914 replied:
???
no. You don't do `variable SECONDS`, you do `10 SECONDS`.
You also don't divide by 10 because that converts it into miliseconds.

Gurkenglas replied:
Would `(world.time / (1 SECONDS))` as Mothblocks said be better?

Watermelon914 replied:
:shrug:

Mothblocks replied:
`world.time SECONDS` is incorrect because `world.time` is *already* in deicseconds. `SECONDS` multiplies it by 10.

Gurkenglas commented:
(though i would have liked dividing by second instead, integers are easier to do cheap shenanigans with)

Gurkenglas committed:
have your nonmacro

also darn multiplying would have been centiseconds. though i would have liked that more, integers are easier to do cheap shenanigans with

Gurkenglas committed:
remacro
